shell/platform/android/io/flutter/embedding/engine/systemchannels/RestorationChannel.java

```diff
@@ -0,0 +1,111 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+package io.flutter.embedding.engine.systemchannels;
+
+import android.support.annotation.NonNull;
+import io.flutter.Log;
+import io.flutter.embedding.engine.dart.DartExecutor;
+import io.flutter.plugin.common.MethodCall;
+import io.flutter.plugin.common.MethodChannel;
+import io.flutter.plugin.common.StandardMethodCodec;
+
+/**
+ * System channel to exchange restoration data between framework and engine.
+ *
+ * <p>The engine can obtain the current restoration data from the framework via this channel to
+ * store it on disk and - when the app is relaunched - provide the stored data back to the framework
+ * to recreate the original state of the app.
+ */
+public class RestorationChannel {
+  private static final String TAG = "RestorationChannel";
+
+  public RestorationChannel(
+      @NonNull DartExecutor dartExecutor, @NonNull boolean waitForRestorationData) {
+    this.waitForRestorationData = waitForRestorationData;
+    MethodChannel channel =
+        new MethodChannel(dartExecutor, "flutter/restoration", StandardMethodCodec.INSTANCE);
+    channel.setMethodCallHandler(handler);
+  }
+
+  /**
+   * Whether {@code setRestorationData} will be called to provide restoration data for the
+   * framework.
+   *
+   * <p>When this is set to true, the channel will delay answering any requests for restoration data
+   * by the framework until {@code setRestorationData} has been called. It must be set to false if
+   * the engine never calls {@code setRestorationData}. If it has been set to true, but it later
+   * turns out that there is no restoration data, {@code setRestorationData} must be called with
+   * null.
```

Comment:
this doc is good. I'd add a bit more explanations for the supply side. i.e. I'm constructing an engine. Why should I set this to true or false?

Reply:
Added another sentence. Hope that helps.